pkg/initializer/initializer.go

```diff
@@ -48,15 +47,12 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 	ctx := context.Background()
 	var err error
 
-	podName, err := miscellaneous.GetEnvVarOrError("POD_NAME")
 	if err != nil {
 		logger.Fatalf("Error reading POD_NAME env var : %v", err)
 	}
 
 	// Etcd cluster scale-up case
-	// Note: first member of etcd cluster can never be part of scale-up case.
-	// TODO: consider removing this special check for first cluster member when backup-restore can check presence of any member in cluster.
-	if miscellaneous.IsMultiNode(logger) && !strings.HasSuffix(podName, "0") {
+	if miscellaneous.IsMultiNode(logger) {
```

Comment:
Could you please elaborate on how this solution addresses the problem without impacting the scale-up scenarios?

Reply:
this won't cause any issue on scale-up as now scale-up is being detected by 2 things:
i) member list api call
ii) member lease update

Basically backup-restore were skipping these checks for zeroth pod, due to that it directly moves to validation check where it failed to detect the single member restoration.
With this PR, backup-restore will still checks the scale-up condition like `lease-update of member`, hence it can infer that lease was already updated hence it's not a scale-up case but data-dir is not valid hence it's a case of single member restoration.